### PR TITLE
feat(Stats): Add `manifestGapCount` to stats

### LINKED ISSUE #6789: Add `manifestGapCount` to player stats
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes.

**Is your feature request related to a problem? Please describe.**

For the sake of telemetry, we are trying to track how many gaps we were able to detect on a manifest level.

**Describe the solution you'd like**

Add `manifestGapCount` to `shaka.extern.Stats` object.
Add `gapCount` to `shaka.extern.Manifest`, so we can pass this value from `Manifest` to `Stats` once manifest is parsed.
In DASH, it should count how many times there is a non-zero distance between periods.
In HLS, it should represent number of occurrences of `EXT-X-GAP`.
In MSS, set it to 0 (?).
In src= mode it should be set to `NaN` as we are not able to determine it.

**Describe alternatives you've considered**

**Additional context**

**Are you planning send a PR to add it?**
Yes if there's an agreeement.

### PR DESCRIPTION
**Pull request by tykus160**
Resolves #6789

### CODE REVIEW

File: externs/shaka/manifest.js [hunk 1]
```diff
@@ -102,6 +103,11 @@
  *   Period elements in a manifest. If streaming protocol does not implement
  *   period-like structure, it should be set to 1.
  *   <i>Defaults to <code>1</code>.</i>
+ * @property {number} gapCount
+ *   The amount of gaps found in a manifest. For DASH, it represents number of
+ *   discontinuities found between periods. For HLS, it is a number of EXT-X-GAP
```
Comment (avelad) [inline externs/shaka/manifest.js:108]: You should add `GAP=YES`.

File: externs/shaka/player.js [hunk 2]
```diff
@@ -180,6 +181,11 @@ shaka.extern.StateChange;
  *   The amount of periods occurred in the manifest. For DASH it represents
  *   number of Period elements in a manifest. For HLS & MSS it is always 1.
  *   In src= mode or if nothing is loaded, NaN.
+ * @property {number} manifestGapCount
+ *   The amount of gaps found in a manifest. For DASH, it represents number of
+ *   discontinuities found between periods. For HLS, it is a number of EXT-X-GAP
```
Comment (avelad) [inline externs/shaka/player.js:186]: You should add `GAP=YES`.

**Review by avelad — CHANGES_REQUESTED** [review]

File: externs/shaka/manifest.js (hunk 1 above)
Comment (tykus160) [reply externs/shaka/manifest.js:108]: Done!

File: externs/shaka/player.js (hunk 2 above)
Comment (tykus160) [reply externs/shaka/player.js:186]: Done!

**Review by avelad — APPROVED** [review]

**Review by theodab — APPROVED** [review]

**Pull request merged**